packages/web3-eth-accounts/src/index.js

```diff
@@ -264,18 +271,20 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
 
     // Otherwise, get the missing info from the Ethereum Node
     return Promise.all([
-        isNot(tx.chainId) ? _this._ethereumCall.getChainId() : tx.chainId,
+        ((isNot(tx.common) || isNot(tx.common.customChain.chainId)) ? //tx.common.customChain.chainId is not optional inside tx.common if tx.common is provided
+            ( isNot(tx.chainId) ? _this._ethereumCall.getChainId() : tx.chainId) 
+            : undefined ), 
```

Comment:
Why we need to specify an `undefined` value in the promise chain? the `chainId` must exists either in the tx object or from the ethereum call.

Reply:
Technically ChainId is optional. 

But this logic is already adding it, via tx.chainId or from common obj, PR was all about avoiding "the chainId must exists either in the tx object or from the ethereum call." in case common is there ;)